src/App.js

```diff
@@ -35,13 +33,39 @@ function App() {
   }
 
   function updateQuantity(itemId, quantity) {
-    const updatedItems = menuItems.map((item) => {
-      if (itemId === item.id) {
-        item.quantity = quantity;
-      }
-      return item;
+    const filteredOrder = order.filter((orderItem) => {
+      return itemId === orderItem.id;
     });
-    setMenuItems(updatedItems);
+    if (filteredOrder.length === 0) {
+      //add item, because id doesnt exist
+      const updatedOrder = [
+        ...order,
+        {
+          id: itemId,
+          quantity: quantity,
+        },
+      ];
```

Comment:
couldn't you add the item and update the quantity in one go?

Reply:
I bet I could but mostly I need to do several small steps to understand what I need to do.. :D I can optimize it later. thx!